src/runtime/hip/hip_hardware_manager.cpp

```diff
@@ -258,6 +258,18 @@ hip_hardware_context::get_property(device_uint_property prop) const {
   case device_uint_property::max_compute_units:
     return _properties->multiProcessorCount;
     break;
+    case device_uint_property::max_work_group_range0:
+    return _properties->maxGridSize[0];
+    break;
+  case device_uint_property::max_work_group_range1:
+    return _properties->maxGridSize[1];
+    break;
+  case device_uint_property::max_work_group_range2:
+    return _properties->maxGridSize[2];
+    break;
+  case device_uint_property::max_work_group_range_size:
+    return std::numeric_limits<std::size_t>::max();
```

Comment:
I'm not sure if it might not be better to effectively do something like `min(maxGridSize[0]*maxGridSize[1]*maxGridSize[2], numeric_limits<size_t>::max())` here (with some handling of potential overflow in the multiplication). Similarly for other backends that don't have direct support for the query.

It seems a little counterintuitive to say that the number of groups is limited in all dimensions (presumably much stricter limits than `numeric_limits<size_t>::max()`), but the overall number of work groups is unlimited.

The wording in the current KHR proposal implies to me that it should always hold that `maxGridSize[0]*maxGridSize[1]*maxGridSize[2] >= max_work_group_range_size`, but maybe I'm reading it wrong (perhaps this should also be clarified in the KHR).

Reply:
Okay, Greg thinks this is fine as is :) https://github.com/KhronosGroup/SYCL-Docs/pull/712#issuecomment-2980288008